sources/pmwebapi_test.go

```diff
@@ -0,0 +1,98 @@
+package sources
+
+import (
+	"testing"
+	"encoding/json"
+)
+
+type AutoGenerated struct {
+	Args struct {
+	} `json:"args"`
+	Headers struct {
+		Accept string `json:"Accept"`
+		AcceptEncoding string `json:"Accept-Encoding"`
+		AcceptLanguage string `json:"Accept-Language"`
+		CacheControl string `json:"Cache-Control"`
+		Connection string `json:"Connection"`
+		Host string `json:"Host"`
+		UpgradeInsecureRequests string `json:"Upgrade-Insecure-Requests"`
+		UserAgent string `json:"User-Agent"`
+		XBluecoatVia string `json:"X-Bluecoat-Via"`
+	} `json:"headers"`
+	Origin string `json:"origin"`
+	URL string `json:"url"`
+}
+
+func TestGetRequest(t *testing.T) {
+	testString := "http://httpbin.org/get"
+	response := getRequest(testString)
+	if response == nil {
+		t.Fatal("retrieved an unexpected result")
```

Comment:
I think all of your t.Fatal() messages need to be more descriptive regarding what the problem is when these failures occur. Most of them just say "fail", which isn't going to help much if 3-4 of them fail at the same time.